bin/test-merges.js

```diff
@@ -0,0 +1,310 @@
+    // fetch any changes from the remote
+    git.fetch({}, (err, data) => {
+      if (err) {
+        reject(err);
+        return;
+      }
+      console.log(`Starting github checkout of ${commitId}...`);
+      git.checkout(commitId, (err, data) => {
+        if (err) {
+          reject(err);
+          return;
+        }
+        resolve(commitId);
+        return;
+      });
+    });
+  });
+}
+
+function exportToLocalMC(commitId) {
+  return new Promise((resolve, reject) => {
+    console.log("Preparing mochitest dev environment...");
+    // Weirdly, /bin/yes causes npm-run-all bundle-static to explode, so we
+    // use echo.
+    shelljs.exec(`echo yes | SYMLINK_TESTS=false bash -x ${PREPARE_MOCHITESTS_DEV}`,
+      {async: true, cwd: TESTING_GIT_AS, silent: false}, (code, stdout, stderr) => {
+        if (code) {
+          reject(new Error(`${PREPARE_MOCHITESTS_DEV} failed, exit code: ${code}`));
+          return;
+        }
+
+        resolve(commitId);
+        return;
+      });
+  });
+}
+
+function commitToHg(commitId) {
+  return new Promise((resolve, reject) => {
+    console.log(`Committing exported ${commitId} to ${AS_REPO_NAME}...`);
+    shelljs.exec(`${HG} commit --addremove -m 'Export of ${commitId} from ${AS_REPO_OWNER}/${AS_REPO_NAME}' .`,
+      {async: true, cwd: TESTING_LOCAL_MC},
+      (code, stdout, stderr) => {
+        if (code) {
+          reject(new Error(`${HG} commit failed, output: ${stderr}`));
+          return;
+        }
+
+        resolve(code);
+        return;
+      }
+    );
+  });
+}
+
+/**
+ * [pushToHgProjectBranch description]
+ *
+ * @return {Promise<String|Number>} resolves with the text written to XXXstdout, or
+ *                                  rejects with the exit code from ${HG}.
+ */
+function pushToHgProjectBranch() {
+  return new Promise((resolve, reject) => {
+    shelljs.exec(`${HG} push -f ${HG_BRANCH_NAME}`, {async: true, cwd: TESTING_LOCAL_MC},
+      (code, stdout, stderr) => {
+        if (code) {
+          reject(new Error(`${HG} failed, exit code: ${code}`));
+          return;
+        }
+
+        // Grab the last linked revision from the push output
+        const rev = stdout.split(/(?:\/rev\/|changeset=)/).slice(-1)[0].split("\n")[0];
+        resolve(`[Treeherder: ${rev}](${TREEHERDER_PREFIX}${rev})`);
+        return;
+      }
+    );
+  });
+}
+
+/**
+ * Remove last commit from the repo so the next artifact build will work right
+ */
+function stripTipFromHg() {
+  return new Promise((resolve, reject) => {
+    console.log("Stripping tip commit from mozilla-central so the next artifact build will work ...");
+    shelljs.exec(`${HG} strip --force --rev -1`,
+      {async: true, cwd: TESTING_LOCAL_MC},
+      (code, stdout, stderr) => {
+        if (code) {
+          reject(new Error(`${HG} strip failed, output: ${stderr}`));
+          return;
+        }
+
+        resolve(code);
+        return;
```

Comment:
They're not, but I find making them more explicit makes it easier to read the code correctly.